upgrading/automated_upgrades.adoc

```diff
@@ -13,32 +20,114 @@
+set of node group definitions looks like in YAML format in the  *_{pb-prefix}roles/openshift_facts/defaults/main.yml_* file:
+
+----
+openshift_node_groups:
+  - name: node-config-master
+    labels:
+      - 'node-role.kubernetes.io/master=true'
+    edits: []
+  - name: node-config-infra
+    labels:
+      - 'node-role.kubernetes.io/infra=true'
+    edits: []
+  - name: node-config-compute
+    labels:
+      - 'node-role.kubernetes.io/compute=true'
+    edits: []
+  - name: node-config-master-infra
+    labels:
+      - 'node-role.kubernetes.io/infra=true,node-role.kubernetes.io/master=true'
+    edits: []
+  - name: node-config-all-in-one
+    labels:
+      - 'node-role.kubernetes.io/infra=true,node-role.kubernetes.io/master=true,node-role.kubernetes.io/compute=true'
+    edits: []
+----
+
+The set of node groups to be used in the cluster must be defined using the
+`openshift_node_groups` variable in the inventory file, however they must be
+first translated into a Python dictionary. You can then use the `edits` field to
+modify any node configuration variables as desired by specifying the key-value
+pairs.
+
+[NOTE]
+====
+See
+xref:../install_config/master_node_configuration.adoc#node-configuration-files[Master
+and Node Configuration Files] for reference on configurable node variables.
+====
+
+For example, the following entry defines a group named `node-config-compute` and
+modifies the node configuration to set `kubeletArguments.pods-per-core` to `20`:
+
+----
+openshift_node_groups=[{'name': 'node-config-compute', 'labels': ['node-role.kubernetes.io/compute=true'], 'edits': [{ 'key': 'kubeletArguments.pods-per-core','value': ['20']}]}]
+----
+
+[[upgrade-workflow]]
+=== Upgrade Workflow
+
+The 3.9 to 3.10 control plane upgrade performs the following steps for
 you:
 
 * A backup of all etcd data is taken for recovery purposes.
-* The API and controllers are updated from 3.7 to 3.8.
-* Internal data structures are updated to 3.8.
-* A second backup of all etcd data is taken for recovery purposes.
-* The API and controllers are updated from 3.8 to 3.9.
-* Internal data structures are updated to 3.9.
-* The default router, if one exists, is updated from 3.7 to 3.9.
-* The default registry, if one exists, is updated from 3.7 to 3.9.
+* The API and controllers are updated from 3.9 to 3.10.
+* Internal data structures are updated to 3.10.
+* The default router, if one exists, is updated from 3.9 to 3.10.
+* The default registry, if one exists, is updated from 3.9 to 3.10.
 * The default image streams and InstantApp templates are updated.
 
-The automated 3.7 to 3.9 node upgrade performs a rolling update of nodes, which:
+The 3.9 to 3.10 node upgrade performs a rolling update of nodes, which:
 
 * Marks a subset of nodes unschedulable and drains them of pods.
-* Updates node components from 3.7 to 3.9 (including openvswitch and container
+* Updates node components from 3.9 to 3.10 (including Open vSwitch and container
```

Comment:
* Updates node components from 3.9 to 3.10
* Converts from local configuration to bootstrapped TLS and Config
* Converts SDN components from systemd services to DaemonSets